src/main/java/task/TaskList.java

```diff
@@ -0,0 +1,120 @@
+package task;
+
+import java.util.ArrayList;
+
+import storage.Storage;
+
+/**
+ * The class handling the Tasklist, a self-initiated data structure list.
+ */
+
+public class TaskList {
+    protected ArrayList<Task> todoList;
+
+    public TaskList(ArrayList<Task> todoList) {
+        this.todoList = todoList;
+    }
+
+    public void add(Task t) {
+        this.todoList.add(t);
+    }
+
+    public void printList() {
+        int length = this.todoList.size();
+        if (length == 0) {
+            System.out.println("There is no task in your list.");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+
+            for (int i = 0; i < length; i++) {
+                String pos = String.valueOf(i + 1);
+                System.out.println(pos + "." + this.todoList.get(i));
+            }
+        }
+    }
+
+    private void showResult(TaskList result, String keyword) {
+        int length = result.todoList.size();
+        if (length == 0) {
+            System.out.println("There is no task in your list that contains the keyword '" + keyword + "'");
+        } else {
+            for (int i = 0; i < length; i++) {
+                String pos = String.valueOf(i + 1);
+                System.out.println(pos + "." + result.todoList.get(i));
+            }
+        }
+    }
+
+    /**
+     * List the whole TodoList after adding any tasks.
+     * 
+     * @param t
```

Comment:
Try to add a description of the param as well

Reply:
Sure